six/modules/c++/six/source/NITFWriteControl.cpp

```diff
@@ -866,11 +866,14 @@ void NITFWriteControl::save(
                 iWriter.setWriteCaching(1);
 
                 nitf::ImageSource iSource;
-                const size_t bandSize = numCols * info.getData()->getNumRows();
+                const NITFSegmentInfo segmentInfo = imageSegments[jj];
+                const size_t bandSize = numCols * segmentInfo.numRows;
 
                 for (size_t chan = 0; chan < numChannels; ++chan)
                 {
-                    nitf::MemorySource ms(imageData[i], bandSize,
+                    nitf::MemorySource ms(imageData[i] +
```

Comment:
It looks like the last parameter has to do with the number of pixels in between columns or something like that. I think I could add the offset to the 4th parameter, but I think it reads a bit more clearly if the offset due to the segment is separate from the offset due to the channel. Everything else is fixed.